src/expressions/literal-expr.md

```diff
@@ -26,29 +26,227 @@ Each of the lexical [literal][literal tokens] forms described earlier can make u
+
+The literal expression's _represented string_ is a sequence of characters derived from the literal content as follows:
+
+* If the token is a [STRING_LITERAL], each escape sequence of any of the following forms occuring in the literal content is replaced by the escape sequence's escaped value.
+    * [Simple escapes]
+    * [7-bit escapes]
+    * [Unicode escapes]
+    * [String continuation escapes]
+
+  These replacements take place in left-to-right order.
+  For example, the token `"\\x41"` is converted to the characters `\` `x` `4` `1`.
+
+* If the token is a [RAW_STRING_LITERAL], the represented string is identical to the literal content.
+
+The expression's value is a reference to a statically allocated [`str`][textual types] containing the UTF-8 encoding of the represented string.
+
+Examples of string literal expressions:
+
+```rust
+"foo"; r"foo";                     // foo
+"\"foo\""; r#""foo""#;             // "foo"
+
+"foo #\"# bar";
+r##"foo #"# bar"##;                // foo #"# bar
+
+"\x52"; "R"; r"R";                 // R
+"\\x52"; r"\x52";                  // \x52
+```
 
 ## Byte literal expressions
 
 A byte literal expression consists of a single [BYTE_LITERAL] token.
 
-> **Note**: This section is incomplete.
+The expression's type is the primitive [`u8`][numeric types] type.
+
+The token must not have a suffix.
+
+The token's _literal content_ is the sequence of characters following the first `U+0027` (`'`) and preceding the last `U+0027` (`'`) in the string representation of the token.
+
+The literal expression's _represented character_ is derived from the literal content as follows:
+
+* If the literal content is one of the following forms of escape sequence, the represented character is the escape sequence's escaped value:
+    * [Simple escapes]
+    * [8-bit escapes]
+
+* Otherwise the represented character is the single character that makes up the literal content.
+
+The expression's value is the represented character's [Unicode scalar value].
+
+> **Note**: the permitted forms of a [BYTE_LITERAL] token ensure that these rules always produce a single character, whose Unicode scalar value is in the range of [`u8`][numeric types].
+
+Examples of byte literal expressions:
+
+```rust
+b'R';                              // 82
+b'\'';                             // 39
+b'\x52';                           // 82
+b'\xA0';                           // 160
+```
 
 ## Byte string literal expressions
 
-A string literal expression consists of a single [BYTE_STRING_LITERAL] or [RAW_BYTE_STRING_LITERAL] token.
+A byte string literal expression consists of a single [BYTE_STRING_LITERAL] or [RAW_BYTE_STRING_LITERAL] token.
 
-> **Note**: This section is incomplete.
+The expression's type is a shared reference (with `static` lifetime) to an array whose element type is [`u8`][numeric types].
+That is, the type is `&'static [u8; N]`, where `N` is the number of characters in the represented string described below.
```

Comment:
Perhaps to avoid confusion here?

```suggestion
That is, the type is `&'static [u8; N]`, where `N` is the number of bytes in the represented string described below.
```

Also, this may become a little more confusing after [RFC 3349](https://rust-lang.github.io/rfcs/3349-mixed-utf8-literals.html) is implemented/stabilized.

Reply:
After RFC 3349 I think the byte and byte-string literals will need a different approach, with a separate version of the whole "Escapes" section. But that will be easier to describe when the new forms of escape are there.